File: client/src/main/java/com/github/dkurata38/open_web_library/client/GoogleCloudVisionClient.java
package com.github.dkurata38.open_web_library.client;

import com.github.dkurata38.open_web_library.application.book.ImageCognitionClient;
import lombok.RequiredArgsConstructor;
import org.springframework.cloud.gcp.vision.CloudVisionTemplate;
import org.springframework.core.io.Resource;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class GoogleCloudVisionClient implements ImageCognitionClient {

	private final CloudVisionTemplate cloudVisionTemplate;

	@Override
	public String extractTextFromImage(Resource resource) {
		return cloudVisionTemplate.extractTextFromImage(resource);
	}
}

File: client/src/main/kotlin/com/github/dkurata38/open_web_library/client/image_congnition/GoogleCloudVisionClient.kt
package com.github.dkurata38.open_web_library.client.image_congnition

import com.github.dkurata38.open_web_library.application.book.ImageCognitionClient
import org.springframework.cloud.gcp.vision.CloudVisionTemplate
import org.springframework.core.io.Resource
import org.springframework.stereotype.Component

@Component
class GoogleCloudVisionClient(private val cloudVisionTemplate: CloudVisionTemplate) : ImageCognitionClient {
	override fun extractTextFromImage(resource: Resource): String {
		return cloudVisionTemplate.extractTextFromImage(resource)
	}
}
